modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/util/CPDefinitionHelperImpl.java

```diff
@@ -213,6 +213,13 @@ private CPDefinitionSearcher _getCPDefinitionSearcher(
 	private String _getFriendlyURL(long cProductId, ThemeDisplay themeDisplay)
 		throws PortalException {
 
+		CPDefinition cpDefinition =
+			_cpDefinitionLocalService.fetchCPDefinitionByCProductId(cProductId);
+
+		if (!cpDefinition.isApproved() || !cpDefinition.isPublished()) {
+			return null;
```

Comment:
You're just returning a null here, did you check that all places where this "result" is used are managing a "null" value?
You should either add a null check there, or returning an empty string here.
cc @andrea-ale-sbarra

Reply:
@luca-pellizzon ,

I resent the PR and introduced the fix in CPDefinitionAssetRenderer.getURLViewInContext() instead. This ensures we don't have to do the check in CPDefinitionHelper and allows us to keep the fix more contained. Also, the method already has null checks being managed in all the places where the "result" is used.